### PR TITLE
Limit module

### PR DESCRIPTION
**Pull request by heswithme**

### CODE REVIEW

File: contracts/DepositLimitModule.vy [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+def available_deposit_limit(receiver: address) -> uint256:
+    """
+    @notice Checks the available deposit limit for a given receiver.
+    @param receiver The address querying deposit limit.
+    @return uint256 Returns the maximum deposit limit if deposits are not paused, otherwise returns 0.
+    """
+    if self.deposits_paused:
+        return 0
+    else:
+        return max_value(uint256)
```
Comment (AlbertoCentonze) [inline contracts/DepositLimitModule.vy:123]: Here you probably want a settable cap, we're not allowing people to go all in from the start

File: contracts/DepositLimitModule.vy [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+################################################################
+
+@deploy
+def __init__():
+    """
+    @notice Initializes the contract by assigning the deployer as the initial admin and controller.
+    """
+    self._set_admin(msg.sender, True)
+    self._set_controller(msg.sender, True)
+    self._set_deposits_paused(False)
```
Comment (AlbertoCentonze) [inline contracts/DepositLimitModule.vy:33]: redundant assignment here

File: contracts/DepositLimitModule.vy [hunk 3]
```diff
@@ -0,0 +1,123 @@
+# pragma version ~=0.4
+
+"""
+@title Deposit Limit Module for yearn vaults
```
Comment (AlbertoCentonze) [inline contracts/DepositLimitModule.vy:4]: You probably want to mention that this module is as temporary measure and will be removed once the vault is battletested

File: tests/unitary/conftest.py [hunk 4]
```diff
@@ -16,10 +16,23 @@ def curve_dao():
...
+@pytest.fixture(scope="module")
+def deposit_limit_controller():
+    return boa.env.generate_address()
+
+
+@pytest.fixture(scope="module")
+def deposit_limit_module(dev_address):
+    contract_deployer = boa.load_partial("contracts/DepositLimitModule.vy")
+    with boa.env.prank(dev_address):
+        contract = contract_deployer()
```
Comment (AlbertoCentonze) [inline tests/unitary/conftest.py:32]: Absolutely irrelevant detail but for one line pranks better to use sender

File: tests/unitary/yearn/test_deposit.py
Comment (AlbertoCentonze) [inline tests/unitary/yearn/test_deposit.py:1]: Since you use the same amount everywhere might be easier to just make a helper var with that amount

**Review by AlbertoCentonze — CHANGES_REQUESTED** [review]

File: contracts/DepositLimitModule.vy (hunk 2 above)
Comment (heswithme) [reply contracts/DepositLimitModule.vy:33]: while it's false by default, i decide to be explicit here

File: contracts/DepositLimitModule.vy (hunk 1 above)
Comment (heswithme) [reply contracts/DepositLimitModule.vy]: changed, review changes

**Pull request merged**